src/DIRAC/WorkloadManagementSystem/Executor/Scouting.py

```diff
@@ -0,0 +1,133 @@
+
+        else:
+            result = jobState.getManifest()
+            if not result['OK']:
+                return S_ERROR('Could not retrieve job manifest: %s' % result['Message'])
+            jobManifest = result['Value']
+            scoutID = jobManifest.getOption('ScoutID', None)
+            if not scoutID:
+                self.jobLog.info('Skipping optimizer, since no scout \
+                                 corresponding to this job group')
+                return self.setNextOptimizer(jobState)
+
+            scoutFlag = 0
+            result = jobState.getAttribute('RescheduleCounter')
+            if not result['OK']:
+                return S_ERROR('Could not retrieve RescheduleCounter')
+            if result['Value'] == None:
+                return S_ERROR('Reschedule Counter not found')
+
+            rCounter = result['Value']
+            if int(rCounter) > 0:
+                rCycle = int(rCounter) - 1
+                result = self.__jobDB.getAtticJobParameters(jid, ['"ScoutFlag"'],
+                                                    rescheduleCounter=rCycle)
+                self.jobLog.info("From AtticJobParameter: %s" % result)
+                if result['OK']:
+                    try:
+                        scoutFlag = result['Value'].get(rCycle).get('ScoutFlag', 0)
+                    except:
+                        pass
+                else:
+                    self.jobLog.info(result['Message'])
+            self.jobLog.info('Setting scoutparams (ID:%s, Flag:%s) to JobParamter'
+                              % (scoutID, scoutFlag))
+            result = self.__setScoutparamsInJobParameters(jid, scoutID, scoutFlag, jobState)
+            if not result['OK']:
+                self.jobLog.info('Skipping, since failed in setting scoutparams of JobParameters.')
+                return self.setNextOptimizer(jobState)
+
+        if int(scoutFlag) == 1:
+            self.jobLog.info('Skipping optimizer, since corresponding scout jobs complete \
+                             (ScoutFlag = %s)'% scoutFlag)
+            return self.setNextOptimizer(jobState)
+
+        self.jobLog.info('Job %s set scouting status' % jid)
+        result = self.__setScoutingStatus(jobState)
+        return result
+
+    def __getIDandFlag(self, scoutparams):
+
+        scoutID = scoutparams.get('ScoutID')
+        scoutFlag = scoutparams.get('ScoutFlag')
+        return scoutID, scoutFlag
+
+    def __setScoutparamsInJobParameters(self, jid, scoutID, scoutFlag, jobState=None):
+
+        if not jobState:
+            jobState = self.__jobData.jobState
+
+        paramList = []
+        paramList.append(('ScoutID', scoutID))
+        paramList.append(('ScoutFlag', scoutFlag))
+        result = self.__jobDB.setJobParameters(jid, paramList)
+        if not result['OK']:
+            self.jobLog.info('Skipping, since failed in recovering scoutparams of JobParameters.')
+            return result
+
+        return S_OK()
```

Comment:
```suggestion
        result = self.__jobDB.setJobParameters(jid, paramList)
        if not result['OK']:
            self.jobLog.info('Skipping, since failed in recovering scoutparams of JobParameters.')

        return result
```